src/pages/research.astro

```diff
@@ -0,0 +1,1131 @@
+                                stroke-linejoin="round"
+                                d="M19 9l-7 7-7-7"
+                              />
+                            </svg>
+                          </label>
+                        )}
+                      </div>
+                    </div>
+                    <div class="abstract-content mt-4 text-sm text-gray-700">
+                      {paper.abstract &&
+                        paper.abstract
+                          .split("\n\n")
+                          .map((paragraph, i) => (
+                            <p class={i > 0 ? "mt-4" : ""}>{paragraph}</p>
+                          ))}
+                    </div>
+                  </article>
+                ))}
+              </div>
+            </section>
+          ))
+        }
+      </div>
+    </Container>
+  </div>
+</Layout>
+
+<style>
+  .btn-action {
+    @apply p-2 text-gray-400 hover:text-gray-600 rounded-full hover:bg-gray-100 transition;
+    padding: 0.5rem;
+    border-radius: 9999px;
+    transition: all 0.3s ease;
+    display: flex;
+    align-items: center;
+    gap: 0.5rem;
+  }
+
+  /* Default styles - buttons always visible */
+  .btn-action .move-button {
+    width: auto;
+    opacity: 1;
+    margin-left: 0.5rem;
+    white-space: nowrap;
+  }
+
+  /* Only apply hover animations on devices that support hover */
+  @media (hover: hover) {
+    .btn-action .move-button {
+      width: 0;
+      overflow: hidden;
+      opacity: 0;
+      transition: all 0.3s ease;
+      margin-left: 0;
+    }
+
+    .paper-entry:hover .btn-action .move-button {
+      width: auto;
+      opacity: 1;
+      margin-left: 0.5rem;
+    }
+  }
+
+  .paper-entry {
+    padding: 1.5rem;
+    background: white;
+    border-radius: 0.5rem;
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05);
+  }
+
+  .paper-entry:hover {
+    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+  }
+
+  .content-container {
+    display: flex;
+    justify-content: space-between;
+    align-items: flex-start;
+  }
+
+  @media (max-width: 640px) {
+    .content-container {
+      flex-direction: column;
+    }
+
+    .buttons-container {
+      margin-left: 0;
+      margin-top: 1rem;
+      width: 100%;
+      justify-content: flex-end;
+    }
+  }
+
+  /* Abstract Content Hidden by Default */
+  .abstract-content {
+    max-height: 0;
+    max-width: 600px;
+    overflow: hidden;
+    transition:
+      max-height 0.3s ease-out,
+      opacity 0.3s ease-out,
+      padding 0.3s ease-out;
+    opacity: 0;
+    text-align: justify;
+    margin: 0 auto;
+  }
+
+  /* When Checkbox is Checked, Show Abstract Content */
+  .abstract-toggle-checkbox:checked ~ .abstract-content {
+    max-height: 2000px; /* Adjust as needed */
+    padding-top: 30px;
+    opacity: 1;
+  }
+  /* Rotate the SVG Icon When Checked */
+  .abstract-toggle-checkbox:checked + div .abstract-toggle svg {
+    transform: rotate(-180deg);
+  }
+</style>
```

Comment:
If you use components, you could apply more tailwind classes, I'd prefer people sticking to tailwind instead of mixing up stuff like this. It's okay to use CSS if it is the last resort, but most of this can be done like the rest of this project does it.

Reply:
Noted. I rarely do UI stuff at all, so I was not aware this is bad style. Maybe I should have looked at how other pages are written.